test/tests/functional/pbs_reservations.py

```diff
@@ -1123,3 +1123,409 @@ def test_reservations_with_expired_subjobs(self):
+                           offset=62, id=jid)
+
+        # Submit a standing reservation, that gets confirmed.  Submission time
+        # is HHMM
+        start = int(time.time()) + 10
+        end = int(time.time()) + 3660
+        a = {ATTR_resv_rrule: 'FREQ=DAILY;COUNT=2',
+             ATTR_resv_timezone: tzone,
+             'reserve_start': start,
+             'reserve_end': end,
+             }
+        r = Reservation(TEST_USER, attrs=a)
+        rid = self.server.submit(r)
+        exp_attr = {'reserve_state': (MATCH_RE, "RESV_CONFIRMED|2")}
+        self.server.expect(RESV, exp_attr, id=rid)
+        rid_q = rid.split(".")[0]
+        # Submit a job-array within resrvation
+        j = Job(TEST_USER, attrs={
+                'Resource_List.walltime': 60, ATTR_q: rid_q, ATTR_J: '1-5'})
+        j.set_sleep_time(60)
+        jid = self.server.submit(j)
+        subjid = []
+        for i in range(1, 6):
+            subjid.append(j.create_subjob_id(jid, i))
+        self.server.expect(JOB, {'job_state': 'Q',
+                                 'comment': (MATCH_RE,
+                                             'Queue not started')}, jid)
+        # Wait for standing reservation first instance to start
+        # Verify one sub-job should running and others in queued
+        exp_attr = {'reserve_state': (MATCH_RE, "RESV_RUNNING|5")}
+        self.server.expect(RESV, exp_attr, id=rid, offset=10)
+        self.server.expect(RESV, {'reserve_index': 1}, id=rid)
+        self.server.expect(JOB, {'job_state': 'B'}, jid)
+        self.server.expect(JOB, {'job_state=R': 1}, count=True,
+                           id=jid, extend='t')
+        self.server.expect(JOB, {'job_state=Q': 4}, count=True,
+                           id=jid, extend='t')
+        # Suspend running sub-job[1] after 10sec, verify
+        # second sub-job start running
+        time.sleep(10)
+        self.server.sigjob(jobid=subjid[0], signal="suspend")
+        self.server.expect(JOB, {'job_state': 'S'}, id=subjid[0])
+        self.server.expect(JOB, {'job_state': 'R'}, id=subjid[1])
+        # Resume suspend sub-job[1], and verify sub-job[1] should
+        # once resource available
+        self.server.sigjob(subjid[0], 'resume')
+        self.server.expect(JOB, {'job_state': 'S'}, id=subjid[0])
+        self.server.delete(subjid[1])
+        self.server.expect(JOB, {'job_state': 'R'}, id=subjid[2])
+        self.server.expect(JOB, {'job_state': 'S'}, id=subjid[0])
+        self.server.delete([subjid[2], subjid[3], subjid[4]])
+        self.server.expect(JOB, {'job_state': 'R'}, id=subjid[0])
+        self.server.expect(JOB, {'job_state': 'F',  'substate': '92',
+                                 'queue': rid_q}, id=subjid[0], extend='x',
+                           offset=61)
+        self.server.expect(JOB, {'job_state': 'F',  'queue': rid_q},
+                           id=jid, extend='x')
+
+    @skipOnCpuSet
+    def test_job_array_within_recurring_reservation(self):
```

Comment:
The name is confusing because the recurring reservation is standing reservation and then it has the same name as the previous test.